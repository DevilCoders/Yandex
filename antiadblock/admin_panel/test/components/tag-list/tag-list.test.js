import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import TagList from 'app/components/tag-list/tag-list';

describe('tag-list component', () => {
    test('render with props', () => {
        const wrapper = shallow(
            <TagList
                value={['value1', 'value2', 'value3']}
                placeholder='placeholder' />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('render with readOnly=true', () => {
        const wrapper = shallow(
            <TagList
                value={['value1', 'value2', 'value3']}
                placeholder='placeholder'
                readOnly />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
