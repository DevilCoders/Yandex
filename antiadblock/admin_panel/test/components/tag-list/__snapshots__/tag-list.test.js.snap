// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tag-list component render with props 1`] = `
<Bem
  block="tag-list"
  mods={
    Object {
      "readOnly": undefined,
      "selected": false,
    }
  }
>
  <Bem
    block="tag-list"
    elem="item"
    key="value1"
  >
    <Bem
      block="tag-list"
      elem="value"
      key="value"
    >
      value1
    </Bem>
    <Bem
      block="tag-list"
      elem="close"
      key="close"
    >
      <button2
        onClick={[Function]}
        tabIndex={0}
        theme="clear"
        view="classic"
      >
        <icon
          type="close"
        />
      </button2>
    </Bem>
  </Bem>
  <Bem
    block="tag-list"
    elem="item"
    key="value2"
  >
    <Bem
      block="tag-list"
      elem="value"
      key="value"
    >
      value2
    </Bem>
    <Bem
      block="tag-list"
      elem="close"
      key="close"
    >
      <button2
        onClick={[Function]}
        tabIndex={0}
        theme="clear"
        view="classic"
      >
        <icon
          type="close"
        />
      </button2>
    </Bem>
  </Bem>
  <Bem
    block="tag-list"
    elem="item"
    key="value3"
  >
    <Bem
      block="tag-list"
      elem="value"
      key="value"
    >
      value3
    </Bem>
    <Bem
      block="tag-list"
      elem="close"
      key="close"
    >
      <button2
        onClick={[Function]}
        tabIndex={0}
        theme="clear"
        view="classic"
      >
        <icon
          type="close"
        />
      </button2>
    </Bem>
  </Bem>
  <Bem
    block="tag-list"
    elem="input"
    key="input"
  >
    <textinput
      controlAttrs={
        Object {
          "onBlur": [Function],
          "onKeyDown": [Function],
          "onKeyUp": [Function],
        }
      }
      innerRef={[Function]}
      onChange={[Function]}
      placeholder="placeholder"
      showListOnFocus={true}
      size="m"
      text=""
      theme="normal"
      tone="grey"
      type="text"
      updateOnEnter={true}
      view="default"
    />
  </Bem>
</Bem>
`;

exports[`tag-list component render with readOnly=true 1`] = `
<Bem
  block="tag-list"
  mods={
    Object {
      "readOnly": true,
      "selected": false,
    }
  }
>
  <Bem
    block="tag-list"
    elem="item"
    key="value1"
  >
    <Bem
      block="tag-list"
      elem="value"
      key="value"
    >
      value1
    </Bem>
  </Bem>
  <Bem
    block="tag-list"
    elem="item"
    key="value2"
  >
    <Bem
      block="tag-list"
      elem="value"
      key="value"
    >
      value2
    </Bem>
  </Bem>
  <Bem
    block="tag-list"
    elem="item"
    key="value3"
  >
    <Bem
      block="tag-list"
      elem="value"
      key="value"
    >
      value3
    </Bem>
  </Bem>
  <Bem
    block="tag-list"
    elem="input"
    key="input"
  />
</Bem>
`;
